### PR TITLE
Clean up WTT stats page

### LINKED ISSUE #163: Totally redesign responsiveness stats pages
### Essentials

* [ ] Make the whole thing less sucky.
* [ ] If we link to the stats page, do people still see the tab with the summary on it?
* [ ] Is the data and methodology clear enough?
* [ ] Can we make the tweet button more prominent/enticing and add a FB button?

### Optional Goodies

* [ ] Include "This MP was new in this Parliament" indicator
* [ ] Include movement indicator

### PR DESCRIPTION
**Pull request by zarino**
Suggested improvements:

- [x] Make it clearer when a column has been sorted.
- [x] Swap order of columns so that response rate isn't at the far right edge.
- [x] Sort out the hierarchy of the search box and intro text on MP league table page.
- [x] Reduce the height of all the stuff before the league table – can we get the table starting within the first 600px of the page? (768px screen, minus ~150px for OS/browser chrome)
- <del>OpenGraph tags should be tailored for the MP results, rather than the general page, once you've searched for an MP.</del> – Moved to https://github.com/mysociety/writetothem/issues/319
- <del>Once you've searched for an MP by postcode (and the MP results are on screen), the various social sharing buttons on the page should link to the page *with* that MP’s details pulled out. Rather than just linking to the generic page.</del> – Moved to https://github.com/mysociety/writetothem/issues/320
- <del>Add ability to search by MP name, rather than just by postcode.</del> – Moved to https://github.com/mysociety/writetothem/issues/321
- [ ] ?? Use percentages for the response rate, rather than "fuzzed" words.

Fixes #163.

Demo version here: http://zarino.writetothem.dev.mysociety.org/stats/2014/mps

screen shot 2016-05-16 at 14 43 39

### CODE REVIEW

**Comment by wrightmartin** [comment]
I feel like this section needs some work - heirarchy or something...

screen shot 2016-05-13 at 08 14 52

**Comment by abimysoc** [comment]
> Social sharing buttons should link to the current URL (including postcode)

Careful. People may not want to disclose their (presumably home) postcode on social media.

**Comment by zarino** [comment]
> Careful. People may not want to disclose their (presumably home) postcode on social media.

@struan @dracos – Is it possible to link to an [MP result page like this](https://www.writetothem.com/stats/2013/mps?pc=w1a+1aa), except by providing some identifier about the MP as a URL parameter, rather than the user’s postcode?

**Comment by struan** [comment]
Yes, I think we could do that. They would need to look up their MP by postcode but that gives us their parlparse id and we could add an ?mpid= param option fairly quickly and include a "share this on $social_media" type think with that, or a "link to this MPs stats" link.

**Comment by zarino** [comment]
@struan: I've finished this branch off, and squashed it into something more logical. Can you merge and deploy?

File: web/stats.php [hunk 1]
```diff
@@ -90,9 +91,13 @@
...
 $area_representatives = array();
-if ($postcode) {
+$rep_info = array();
+if ( $parlparse_id) {
+    $rep_info['parlparse_person_id'] = 'uk.org.publicwhip/person/' . $parlparse_id;
+    $rep_info['postcode'] = '';
+} else if ($postcode) {
     $area_name = '';
-    $voting_areas = mapit_call('postcode', $postcode, $mapit_args, array(
+    $voting_areas = mapit_call('postcode', $postcode, array(), array(
```
Comment (dracos) [inline web/stats.php]: Not sure if this is meant to go?

File: web/stats.php [hunk 2]
```diff
@@ -90,9 +91,13 @@
 
 $error_message = '';
 $area_representatives = array();
-if ($postcode) {
+$rep_info = array();
```
Comment (dracos) [inline web/stats.php]: This is now getting cleared again later on.

File: templates/website/stats-mp-performance.html [hunk 3]
```diff
@@ -156,11 +160,13 @@ <h2 class="big-title"><?=$values['title']?></h2>
                         </p>
                     </div>
                     <div class="columns small-4">
+                        <?php if ($mp['pc']) { ?>
```
Comment (dracos) [inline templates/website/stats-mp-performance.html]: This should just be $pc I think, now.

File: web/stats.php [hunk 4]
```diff
@@ -90,9 +91,13 @@
...
+if ( $parlparse_id) {
+    $rep_info['parlparse_person_id'] = 'uk.org.publicwhip/person/' . $parlparse_id;
+    $rep_info['postcode'] = '';
```
Comment (dracos) [inline web/stats.php]: Don't think we need rep_info postcode any more, if the template uses $pc.

**Pull request merged**